source/agora/common/Block.d

```diff
@@ -130,10 +147,33 @@ public struct Block
     public void serialize (scope SerializeDg dg) pure const nothrow @safe
     {
         serializePart(this.header, dg);
+        serializePart(this.txs.length, dg);
         foreach (ref tx; this.txs)
             serializePart(tx, dg);
     }
 
+    /***************************************************************************
+
+        Block Deserialization
+
+        Params:
+            dg = deserialize function accumulator
+
+    ***************************************************************************/
+
+    public void deserialize (scope DeserializeDg dg) nothrow @safe
+    {
+        deserializePart(this.header, dg);
+        ulong size;
+        deserializePart(size, dg);
+        this.txs.length = size;
+        foreach (idx; 0 .. size)
+        {
+            deserializePart(this.txs[idx], dg);
+        }
+        this.buildMerkleTree();
```

Comment:
This is going to be expensive on the long run

Reply:
add on Serialize MekleTree[] & deserialize